### DISCUSSION TITLE
Docker push run very long time

### DISCUSSION THREAD

**Discussion by i-kostyakov** [community/community#26416] [Actions]
<p>Hi to all.<br>
I got a trouble, docker push action in my workflow running a very long time (and didn’t finish). In the end I canceled this run after 1 hour.<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/7/e/7e52107af6beebf0f69591315730300fd7e57607.png" data-download-href="/uploads/short-url/i1tYY9OMNepwJmOdEpamKRX59qL.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/35257530/181097572-7c499087-431d-4cc9-9871-47ab96ac6b48.png" alt="image" data-base62-sha1="i1tYY9OMNepwJmOdEpamKRX59qL" width="690" height="323" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/7/e/7e52107af6beebf0f69591315730300fd7e57607_2_690x323.png, https://aws1.discourse-cdn.com/github/optimized/3X/7/e/7e52107af6beebf0f69591315730300fd7e57607_2_1035x484.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/7/e/7e52107af6beebf0f69591315730300fd7e57607_2_1380x646.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/7/e/7e52107af6beebf0f69591315730300fd7e57607_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1825×856 40.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>My previous build (8 days ago) worked fine and pushed image to registry in 37s</p>
<p>Workflow didn’t change between runs.</p>
<p>My workflow:</p>
<pre><code class="lang-yaml">name: Staging

on:
  push:
    branches: [ master ]

jobs:
  delivery:
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@v2
      - name: Docker login
        run: echo ${{ secrets.DOCKER_PASSWORD }} | docker login ${{ secrets.DOCKER_REGISTRY }} -u ${{ secrets.DOCKER_USER }} --password-stdin
      - name: Build
        run: docker build -t back .
      - name: Tags
        run: docker tag back ${{ secrets.DOCKER_REGISTRY }}/rise-up-back:staging
      - name: Push
        run: docker push ${{ secrets.DOCKER_REGISTRY }}/rise-up-back:staging
      - name: Deploy to staging
        uses: appleboy/ssh-action@master
        with:
          host: ${{ secrets.STAGING_HOST }}
          username: ${{ secrets.STAGING_USER }}
          key: ${{ secrets.STAGING_SSH_KEY }}
          script: |
            cd projects/new-matrix-backend/
            echo ${{ secrets.DOCKER_PASSWORD }} | docker login ${{ secrets.DOCKER_REGISTRY }} -u ${{ secrets.DOCKER_USER }} --password-stdin
            TAG=staging docker-compose -p staging -f docker-compose.prod.yml pull
            TAG=staging docker-compose -p staging -f docker-compose.prod.yml up --no-deps -d main
            docker logout
</code></pre>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/i-kostyakov">@i-kostyakov</a>,</p>
<p>You can try to check with the following steps:</p>
<ul>
<li>
<p><a href="https://docs.github.com/en/free-pro-team@latest/actions/managing-workflow-runs/enabling-debug-logging" rel="noopener nofollow ugc">Enable debug logging</a> to see if you can get more detailed information about the problem from the debug logs.</p>
</li>
<li>
<p>Try to execute these build, tag and push the image on your local machine to see if it can work fine.</p>
</li>
<li>
<p>Try using a self-hosted runner to see if the workflow can work smoothly.</p>
</li>
</ul>

**Comment by i-kostyakov** [comment]
<p>Thank you for your answer.<br>
All works fine on my local machine.<br>
I don’t know what happened on thursday and friday, but today all works fine.</p>
<p>Didn’t know about ability debug logging in GA. Thanks a lot!</p>
